editor/src/components/canvas/commands/wrap-in-conditional-command.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  emptyComments,
+  jsExpressionValue,
+  jsxConditionalExpression,
+  JSXElementChild,
+  jsxFragment,
+} from '../../../core/shared/element-template'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import {
+  EditorState,
+  EditorStatePatch,
+  forUnderlyingTargetFromEditorState,
+  insertElementAtPath,
+  removeElementAtPath,
+} from '../../editor/store/editor-state'
+import { BaseCommand, CommandFunction, getPatchForComponentChange, WhenToRun } from './commands'
+import * as EP from '../../../core/shared/element-path'
+import { getUtopiaJSXComponentsFromSuccess } from '../../../core/model/project-file-utils'
+import { InsertionSubjectWrapper } from '../../editor/editor-modes'
+import { assertNever } from '../../../core/shared/utils'
+
+type ContainerToWrapIn = InsertionSubjectWrapper
+
+export interface WrapInContainerCommand extends BaseCommand {
+  type: 'WRAP_IN_CONDITIONAL'
+  whenToRun: WhenToRun
+  target: ElementPath
+  wrapperUID: string
+  wrapper: ContainerToWrapIn
+}
+
+export function wrapInContainerCommand(
+  whenToRun: WhenToRun,
+  target: ElementPath,
+  wrapperUID: string,
+  wrapper: ContainerToWrapIn,
+): WrapInContainerCommand {
+  return {
+    type: 'WRAP_IN_CONDITIONAL',
+    whenToRun: whenToRun,
+    target: target,
+    wrapperUID: wrapperUID,
+    wrapper: wrapper,
+  }
+}
+
+export const runWrapInConditionalCommand: CommandFunction<WrapInContainerCommand> = (
```

Comment:
This (plus the `type` field and filename) need to be renamed

Reply:
thanks! updated in https://github.com/concrete-utopia/utopia/pull/3531/commits/7b093a70c94b5a663a264a33d734f830bc2e43c6